### PR TITLE
Product Query: Polishing touches for the feature plugin debut

### LINKED ISSUE #7758: Product Query [3]: Polish the block presentation
As we are readying to make the Product Query block available to Feature Plugin users, here are a few tasks we should accomplish:

- [x] Add a (Beta) label after the block name.
- [x] Decide on a user-friendly block name. Current candidate is just: `Products`.
- [x] Change the block icon.
- [x] Add feedback block in the inspector with link to [Airtable](https://href.li/?https://airtable.com/shrFX5FAqmCY6hVYI).
- [ ] (Maybe?) Add a notice in the block description mentioning the block is experimental and going through changes.
- [x] Change the dev flag to a feature plugin flag.
- [x] Add all currently available product elements as the default template.

cc @vivialice

### PR DESCRIPTION
**Pull request by sunyatasattva**
This PR adds the polishing touches mentioned in #7758 to get the Product Query block ready for its feature plugin debut. In particular, it:

* Add a (Beta) label after the block name.
* Change the name to `Products`.
* Change the block icon to stacks.
* Change the block description. 
* Add feedback block in the inspector with link to [Airtable](https://href.li/?https://airtable.com/shrFX5FAqmCY6hVYI).
* Adds all currently available atomic blocks to the default template.
* Change the dev flag to a feature plugin flag.

All of this with some caveats:

* The description still includes the text “Create a new post for this feed”: this is currently hard-coded in the Query Loop block.
* The feedback block doesn't get added to the very bottom. This behavior appears consistent with other blocks (the “Advanced” tools panel appear always at the bottom), but it doesn't seem to be intended in the code of the `FeedbackPrompt` (discussion pending).

Fixes #7758

#### Other Checks

- [x] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [x] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).

### Screenshots

Screenshot 2022-12-01 at 22 50 25
Screenshot 2022-12-01 at 22 49 40

#### User Facing Testing

1. Build the project (or download the `zip` file provided in the PR) without the development flags.
2. Open the inserter in the editor.
2. Make sure the block appears as “Products (Beta)“ instead of “Product Query” and has a “Stacks” icon instead of “Sparkles” icon. Icon should still be purple.
3. Add the block and select it.
4. Make sure the description says: “A block that displays a selection of products in your store.”
5. Make sure there is a Feedback prompt at the bottom of the inspector with a link going [here](https://href.li/?https://airtable.com/shrFX5FAqmCY6hVYI).
6. Make sure the default inner blocks are as follows:

- Product Image
- On-Sale Badge
- Product Stock Indicator
- Post Title
- Product SKU
- Product Price
- Product Rating
- Add to Cart Button
- Product categories
- Product tags

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental

### Performance

Performance is not great. Because each of the atomic blocks is designed to do fetch its own data, having a starting template with all the available blocks creates a lot of queued requests that take a bunch of time to resolve.

We should either have a smaller starting template for now, but ideally we would change the logic of the atomic blocks to be passed the data and context down, instead of fetching their own data. 

### Changelog

> Products block: make the block available to the users of the feature plugin.

### CODE REVIEW

**Comment by vivialice** [comment]
👋  Hi @sunyatasattva! Great work on getting this PR up so quickly. I just did a test of the block. Feedback summarized below: 

**Feedback prompt** 
Here is what I would suggest for the copy.

Headline: `Share your feedback!` 
Body copy: `The Products Block is currently in beta. Your feedback is valuable in helping us develop it.`
CTA: `Give feedback` 

Are we also able to use a button instead? I don't want to add extra work, but if it's a quick swap it will be more noticeable than the link. Here is a mockup: 

Screenshot 2022-12-02 at 11 04 59

In terms of layout, I noticed it appears in the wrong place which you have already highlighted. I would say this needs fixing before going out. 

**Product Elements**
When there are no tags, a note is displayed but when there are no ratings, it doesn't behave the same. I think when an element is unavailable we should choose one direction. I am leaning towards just not including the element and in the case with product tags, removing this entirely from the default. I'd say for the default we should priotize: 

- product image 
- product title
- sale badge (the one that appears with the product image) 
- stock status
- price 
- add to cart 
- categories 
- product rating 

and leave out 
- product sku 
- sale badge 
- product tags 

I also think it would be great if we could spend some time tweaking the sizing and hierarchical structure of how the blocks appear within the tile. I also see your concern about performance and I think we could reduce the above list of loaded elements further to: 

- product image 
- product title 
- sale badge (the one that appears with the product image) 
- price 
- add to cart button

**No Results**
We haven't talked about this, but I am wondering what options we might have to improve this? I feel like it's unclear to the user what they are meant to do with this and was imagining we could create a default for when there are no results. I wonder if we can remove it for now?

File: assets/js/blocks/product-query/constants.ts
Comment (nerrad) [inline assets/js/blocks/product-query/constants.ts]: Let's change the order of these as @vivialice  pointed out in her comment. I agree with what she suggests for removal as well as the note about just using the On Sale badge in the product image. So you'd end up with:

- `woocommerce/product-image`
- `core/post-title` (until this is a variation?)
- `woocommerce/product-price`
- `woocommerce/product-button`

File: assets/js/editor-components/feedback-prompt/index.js
Comment (nerrad) [inline assets/js/editor-components/feedback-prompt/index.js]: You can drop the `https://href.li/?` from the URL here.

**Review by nerrad — CHANGES_REQUESTED** [review]
I've left some comments inline as well.

- @vivialice let's avoid any redesign of the feedback prompt for now. We're using the same component as what is used for the Cart and Checkout blocks. Could maybe collaborate with @elizaan36 about potential holistic changes to the component.
- **Feedback prompt:** Viv's suggestion could be better I think. "Thanks for trying out this block! Help us make it better by sharing your feedback."
- Regarding the layout of the feedback prompt, it showed up for me just above the color panel. I don't think we were ever able to figure out how to get it to render below the Advanced Panel but you could try waiting for an additional render (via the main component) to see if that'll put it below the color panel. If not, don't worry about it. 
- Regarding No Results: @vivialice this is something that is provided by the WP Query Loop block and I think we should leave it for now. It will be useful when the Products block is used in combination with filter blocks and shoppers use filters that result in no results. I think any improvements for this would be good to suggest upstream to the GB project.

**Comment by vivialice** [comment]
> Could maybe collaborate with @elizaan36 about potential holistic changes to the component.

Sure, there are some other ideas for feedback components floating around other teams as well so it might be good for all of us to align on some general principles around them.

**Comment by sunyatasattva** [comment]
Thanks @vivialice and @nerrad for your feedback. I pushed some new changes to the PR addressing it:

### Feedback prompt

I took the liberty to mix a bit both of your suggestions, if you don't mind, to get to a middle ground. I took Darren's text but I took Viv's idea to name the block within it, to make it feel a bit more specific.

I've also added the ability to the `FeedbackPrompt` component to accept different titles and went ahead and used Viv's proposed title.

If that's not ok, please feel free to request other changes with a different copy and I'll change it at once.

Regarding the positioning, @nerrad, the component already waits after the first render to render itself. I tried several other solutions, such as waiting on mounting of the inspector controls, waiting a further re-render, or even waiting very short timeouts. None of that worked.

So I dug on some Gutenberg code: the positioning of advanced controls and also colors/typography/dimension/border is hard-coded in Gutenberg. It'd be nice to have some controls on the befores and afters, either through filters or through more slots, I'll keep that in mind and propose that upstream eventually.

For now, I came up with a hacky solution: it works at least to put it below the Colors panel. With caveats… explained a bit more in the commit message.

Let me know if you think we should go for this hacky solution or revert it. I think it definitely looks a tad better below the colors than above.

### Default template

I went with what @nerrad requested. I totally agree with this more minimal approach (though perhaps I would have also added the rating by default as it is pretty common to see that nowadays), but I wonder: your initial suggestion on adding all the available elements was for discoverability and visibility. What is our plan on that front?

### No results

If need be, “no results” can be removed from the default template as well. I agree with @nerrad that this is still useful in combination with filters, and I think we should leave it by default. The UX for that is very unclear, however, and we should think of some improvements in the near future: either upstream proposals, or we can make a variation (or both: first make a variation, and then after we play with it a bit, propose our changes upstream).

---

Let me know if there is anything else.

File: assets/js/blocks/product-query/inspector-controls.tsx
Comment (nerrad) [inline assets/js/blocks/product-query/inspector-controls.tsx:201]: Given I anticipate this feedback prompt not being around for a while I think this is sufficient for now. When we release the block to WooCommerce core, we'll have to test behaviour against versions of WP that WC Core supports.

**Review by nerrad — APPROVED** [review]
Thanks for your work on this Lucio! From a testing and product review perspective this is good to go.

#### Feedback prompt

The changes look good. I had a minor comment about watching for how the hack works with earlier versions of WP but that won't be until this block surfaces in WC Core. Make sure there is a followup issue to test against earlier versions of WP core and make any necessary adjustments but I don't think it has to block merge.

#### Default template
Thanks for the change

> your initial suggestion on adding all the available elements was for discoverability and visibility. What is our plan on that front?

I still have some concerns about that, but knowing there's a performance impact to having them all there _now_ is why we're reducing. Once we've addressed the performance concerns (I like your idea of sharing the context - there should only really need to be one request to get all product data), we can revisit. Also we still have some work to do on improving the default layout and providing alternatives through patterns.

#### No results 

> The UX for that is very unclear, however, and we should think of some improvements in the near future: either upstream proposals, or we can make a variation (or both: first make a variation, and then after we play with it a bit, propose our changes upstream).

Like this! Let's leave it to be one of the last things though.

**Comment by Aljullu** [comment]
This PR seems to break some tests which rely on the `Product Query` string. I will open a PR on top of this branch to fix them.

**Review by gigitux — COMMENTED** [review]
Thanks for working on this! On the code level, the PR is okay, but I have some questions regarding the UI:
- should the feedback prompt be full-width?
image
- the variations for inner core blocks (e.g: `Post Title`) that we discussed last week will be added by #7836, #7837,  #7838, right?
- ~~I noticed that the filer by Stock Status doesn't work. Is it something that we should address before the release?~~ I merged trunk in this branch and it works 🥳

**Comment by Aljullu** [comment]
Merging without waiting for tests to pass in this PR because they already passed in #7840 and #7833.

**Comment by sunyatasattva** [comment]
> Thanks for working on this! On the code level, the PR is okay, but I have some questions regarding the UI:
> 
> * should the feedback prompt be full-width?

Yes, the feedback prompt should be full-width. Make sure your CSS is reloaded, as sometimes I've got some caching issues with the CSS, and let me know if you still see it like that.

> * the variations for inner core blocks (e.g: `Post Title`) that we discussed last week will be added by #7836, #7837,  #7838, right?

Yup, correct!

**Pull request merged**
